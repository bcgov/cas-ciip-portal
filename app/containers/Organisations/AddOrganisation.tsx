import React, { useState } from "react";
import { graphql, createFragmentContainer, RelayProp } from "react-relay";
import { Dropdown, FormControl, Card, Table } from "react-bootstrap";
import { AddOrganisation_query } from "AddOrganisation_query.graphql";
import AddOrganisationModal from "components/AddOrganisationModal";
import createOrganisationMutation from "mutations/organisation/createOrganisationMutation";
import Organisation from "./Organisation";

interface Props {
  query: AddOrganisation_query;
  relay: RelayProp;
  orgInput: string;
  selectedOrg: number;
  handleInputChange: (event: any) => void;
  handleOrgChange: (event: any) => void;
}

export const AddOrganisationComponent: React.FunctionComponent<Props> = (
  props
) => {
  const [selectedOrgDetails, setSelectedOrgDetails] = useState(null);
  const { allOrganisations } = props.query;
  const changeInput = (event) => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id, tradeName, cra, swrsReportId) => {
    props.handleOrgChange(id);
    props.handleInputChange("");
    setSelectedOrgDetails({ name, id, tradeName, cra, swrsReportId });
  };

  const handleAddOrganisation = async (variables) => {
    const { environment } = props.relay;
    await createOrganisationMutation(environment, variables);
  };

  return (
    <div>
      <Card style={{ marginTop: "50px" }}>
        <Card.Body>
          <Card.Title>Search for an operation: </Card.Title>
          <Dropdown className="search-dropdown">
            <Dropdown.Toggle id="org-dropdown" className="search-toggle">
              Find Operation
            </Dropdown.Toggle>
            <Dropdown.Menu>
              <FormControl
                placeholder="Search..."
                className="mx-3 my-2 w-auto"
                onChange={changeInput}
              />
              <div className="org-scroll">
                {allOrganisations.edges.map(({ node }) => {
                  return (
                    <Organisation
                      key={node.id}
                      select
                      organisation={node}
                      orgInput={props.orgInput}
                      selectOrg={selectOrg}
                    />
                  );
                })}
              </div>
            </Dropdown.Menu>
          </Dropdown>
          {props.selectedOrg !== null && (
            <Table
              style={{
                textAlign: "center",
                margin: "10px 0 2em 0",
                borderBottom: "1px solid rgb(0,0,0,0.1)",
              }}
            >
              <thead>
                <tr>
                  <th>Operator Name</th>
                  <th>Trade Name</th>
                  <th>CRA Business Number</th>
                  <th>SWRS Report Id</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{selectedOrgDetails?.name}</td>
                  <td>{selectedOrgDetails?.tradeName}</td>
                  <td>{selectedOrgDetails?.cra}</td>
                  <td>{selectedOrgDetails?.swrsReportId}</td>
                </tr>
              </tbody>
            </Table>
          )}
          <AddOrganisationModal onAddOrganisation={handleAddOrganisation} />
          <style jsx>
            {`
              .org-scroll {
                max-height: 250px;
                overflow: hidden;
                overflow-y: scroll;
              }

              thead {
                background: #036;
                color: #fff;
              }
            `}
          </style>
        </Card.Body>
      </Card>
    </div>
  );
};

export default createFragmentContainer(AddOrganisationComponent, {
  query: graphql`
    fragment AddOrganisation_query on Query {
      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `,
});
