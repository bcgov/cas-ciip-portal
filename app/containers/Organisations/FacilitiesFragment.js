import React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Card, Accordion } from 'react-bootstrap';

const FacilitiesFragment = (props) => {
    return (<Accordion.Collapse eventKey={props.facility.organisationId}><Card.Body data-testid="card" style={{ borderBottom: '1px solid #d3d3d3' }}>
        <div style={{ fontWeight: 'bold' }}>
            {props.facility.facilityName}

            <Button variant="primary" style={{ marginLeft: '30px' }}>
                Apply for CIIP for this facility
      </Button>
        </div>
        <p style={{ marginTop: '10px' }}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor.
    </p>
    </Card.Body></Accordion.Collapse>);
}

// export default FacilitiesFragment;
export default createFragmentContainer(FacilitiesFragment, {
    facility: graphql`
    fragment FacilitiesFragment_facility on Facility{
        id
        rowId
        organisationId
        facilityName
    }`
});