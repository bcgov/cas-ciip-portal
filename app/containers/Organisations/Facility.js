import React from 'react';
import {graphql, commitMutation, createFragmentContainer} from 'react-relay';
import {Button} from 'react-bootstrap';
import {useRouter} from 'next/router';
import Link from 'next/link';

export const FacilityComponent = ({relay, facility}) => {
  const {environment} = relay;
  const createApplication = graphql`
    mutation FacilityApplicationMutation($input: CreateApplicationInput!) {
      createApplication(input: $input) {
        application {
          id
        }
      }
    }
  `;
  const router = useRouter();
  const startApplication = () => {
    const variables = {
      input: {
        application: {
          facilityId: facility.rowId
        }
      }
    };
    const mutation = createApplication;
    commitMutation(environment, {
      mutation,
      variables,
      onCompleted: response => {
        router.push({
          pathname: '/ciip-application',
          query: {
            applicationId: response.createApplication.application.id
          }
        });
      }
    });
  };

  const {applicationsByFacilityId = {}} = facility;
  const {edges = []} = applicationsByFacilityId;
  const {node = {}} = edges[0] || {};
  const {id: applicationId} = node;

  return (
    <div style={{fontWeight: 'bold'}}>
      {facility.facilityName}

      {applicationId ? (
        <Link
          href={{
            pathname: '/ciip-application',
            query: {
              applicationId
            }
          }}
        >
          <Button variant="primary" style={{marginLeft: '30px'}}>
            Resume CIIP application
          </Button>
        </Link>
      ) : (
        <Button
          variant="primary"
          style={{marginLeft: '30px'}}
          onClick={startApplication}
        >
          Apply for CIIP
        </Button>
      )}
    </div>
  );
};

export default createFragmentContainer(FacilityComponent, {
  facility: graphql`
    fragment Facility_facility on Facility {
      facilityName
      rowId
      applicationsByFacilityId {
        edges {
          node {
            id
          }
        }
      }
    }
  `
});
