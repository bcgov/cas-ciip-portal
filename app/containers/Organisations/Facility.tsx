import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Button, Badge, Card, ListGroup, ListGroupItem} from 'react-bootstrap';
import {useRouter} from 'next/router';
import Link from 'next/link';
import createApplicationMutation from '../../mutations/application/createApplicationMutation';

export const FacilityComponent = ({relay, facility}) => {
  const {environment} = relay;
  const router = useRouter();

  const startApplication = async () => {
    const variables = {
      input: {
        facilityIdInput: facility.rowId
      }
    };

    const response = await createApplicationMutation(environment, variables);
    console.log(response);
    router.push({
      pathname: facility.hasSwrsReport
        ? '/ciip-application-swrs-import'
        : '/ciip-application',
      query: {
        applicationId: response.createApplicationMutationChain.application.id
      }
    });
  };

  const {applicationsByFacilityId = {}} = facility;
  const {edges = []} = applicationsByFacilityId;
  const {node = {}} = edges[0] || {};
  const {id: applicationId, applicationStatus} = node;

  // Conditionall render apply / resume button depending on existence and status of Facility's application
  const applyButton = () => {
    if (!applicationId) {
      return (
        <Button variant="primary" onClick={startApplication}>
          Apply for CIIP for this facility
        </Button>
      );
    }

    if (applicationId && applicationStatus.applicationStatus === 'draft') {
      return (
        <Link
          href={{
            pathname: '/ciip-application',
            query: {
              applicationId
            }
          }}
        >
          <Button variant="primary">Resume CIIP application</Button>
        </Link>
      );
    }

    return null;
  };

  const statusBadgeColor = {
    attention: 'warning',
    pending: 'info',
    declined: 'danger',
    approved: 'success',
    draft: 'dark'
  };

  return (
    <>
      <Card style={{maxWidth: '400px'}}>
        <Card.Header>
          <strong>Facility Name: </strong>
          {facility.facilityName}
        </Card.Header>
        <ListGroup className="list-group-flush">
          <ListGroupItem>
            <strong>Address: </strong> <br />
            {facility.facilityMailingAddress} {facility.facilityPostalCode}
            <br />
            {facility.facilityCity}, {facility.facilityProvince}
          </ListGroupItem>
          <ListGroupItem>
            <strong>Application Status:</strong> &nbsp;{' '}
            {edges.length > 0 ? (
              <>
                <Badge
                  pill
                  variant={
                    statusBadgeColor[applicationStatus.applicationStatus]
                  }
                >
                  {applicationStatus.applicationStatus}
                </Badge>
              </>
            ) : (
              <>Application not started</>
            )}
          </ListGroupItem>
        </ListGroup>
        <Card.Body>{applyButton()}</Card.Body>
      </Card>
    </>
  );
};

export default createFragmentContainer(FacilityComponent, {
  facility: graphql`
    fragment Facility_facility on Facility {
      facilityName
      facilityMailingAddress
      facilityCity
      facilityProvince
      facilityPostalCode
      rowId
      hasSwrsReport(reportingYear: "2018")

      applicationsByFacilityId {
        edges {
          node {
            id
            applicationStatus {
              id
              applicationStatus
            }
          }
        }
      }
    }
  `
});
