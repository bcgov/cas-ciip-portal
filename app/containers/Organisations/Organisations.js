import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {Table, Dropdown, Button, Alert} from 'react-bootstrap';
import Organisation from './Organisation';
import UserOrganisation from './UserOrganisation';

export const OrganisationsComponent = props => {
  const {user, allOrganisations} = props.query;
  if (!user) return '...Loading';

  const changeInput = event => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id) => {
    props.handleInputChange(name);
    props.handleOrgChange(id);
    props.handleContextChange();
  };

  const claimOrg = async () => {
    props.handleContextChange();
    props.handleInputChange('');
    await props.handleOrgConfirm(props.relay.environment);
    props.handleOrgChange(null);
  };

  const cancelClaim = () => {
    props.handleOrgChange(null);
    props.handleContextChange();
    props.handleInputChange('');
  };

  return (
    <>
      {user.userOrganisationsByUserId.edges.length === 0 ? (
        <>
          <Alert variant="warning">
            You are not registered to apply for any Reporting Operation at this
            time. You can request access to a Reporting Operation by selecting
            one from the dropdown below.
          </Alert>
          <br />
        </>
      ) : (
        <Table
          striped
          bordered
          hover
          style={{textAlign: 'left', marginBottom: '40px'}}
        >
          <thead>
            <tr>
              <th>Operation</th>
              <th>Access Status</th>
              <th>Facilities</th>
            </tr>
          </thead>
          <tbody>
            {user.userOrganisationsByUserId.edges.map(({node}) => {
              return <UserOrganisation key={node.id} userOrganisation={node} />;
            })}
          </tbody>
          <style jsx>
            {`
              th {
                padding: 20px;
                font-weight: 500;
              }
            `}
          </style>
        </Table>
      )}

      {props.confirmOrg ? (
        <>
          <h5 className="blue">Requesting access to: </h5>
          <h4 style={{fontWeight: 300, margin: '15px 0'}}>{props.orgInput} </h4>
          <Button
            style={{marginRight: '15px'}}
            variant="primary"
            onClick={claimOrg}
          >
            Request Access
          </Button>
          <Button variant="danger" onClick={cancelClaim}>
            Cancel
          </Button>
        </>
      ) : (
        <Dropdown className="search-dropdown">
          <h5>Add an Operation:</h5>
          <Dropdown.Toggle
            as="input"
            placeholder="Search..."
            onChange={changeInput}
          ></Dropdown.Toggle>
          <Dropdown.Menu>
            {allOrganisations.edges.map(({node}) => {
              return (
                <Organisation
                  key={node.id}
                  select
                  organisation={node}
                  orgInput={props.orgInput}
                  selectOrg={selectOrg}
                />
              );
            })}
          </Dropdown.Menu>
          <style jsx global>{`
            .dropdown-toggle {
              padding: 8px 16px;
              width: 300px;
              border-radius: 3px;
              border: 1px solid #999;
            }
          `}</style>
        </Dropdown>
      )}
    </>
  );
};

export default createFragmentContainer(OrganisationsComponent, {
  query: graphql`
    fragment Organisations_query on Query
      @argumentDefinitions(id: {type: "ID!"}) {
      user(id: "WyJ1c2VycyIsMV0=") {
        id
        userOrganisationsByUserId(first: 2147483647)
          @connection(key: "Organisations_userOrganisationsByUserId") {
          edges {
            node {
              id
              ...UserOrganisation_userOrganisation
            }
          }
        }
      }

      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `
});
