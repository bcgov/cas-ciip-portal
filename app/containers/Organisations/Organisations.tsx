import React from 'react';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {
  Table,
  Dropdown,
  Button,
  FormControl,
  Alert,
  Card
} from 'react-bootstrap';
import {Organisations_query} from 'Organisations_query.graphql';
import {RelayModernEnvironment} from 'relay-runtime/lib/store/RelayModernEnvironment';
import AddOrganisationFacility from 'components/AddOrganisationFacility';
import createOrganisationMutation from 'mutations/organisation/createOrganisationMutation';
import LoadingSpinner from 'components/LoadingSpinner';
import Organisation from './Organisation';
import UserOrganisation from './UserOrganisation';

interface Props {
  query: Organisations_query;
  relay: RelayProp;
  flagCertRequests: boolean;
  orgInput: string;
  selectedOrg: number;
  confirmOrg: boolean;
  handleInputChange: (event: any) => void;
  handleContextChange: () => void;
  handleOrgChange: (event: any) => void;
  handleOrgConfirm: (status: string, env: RelayModernEnvironment) => any;
}

export const OrganisationsComponent: React.FunctionComponent<Props> = (
  props
) => {
  const {session, allOrganisations} = props.query;
  if (!session) return <LoadingSpinner />;

  const changeInput = (event) => {
    event.stopPropagation();
    event.preventDefault();
    event.persist();
    props.handleInputChange(event.target.value);
  };

  const selectOrg = (name, id) => {
    props.handleInputChange(name);
    props.handleOrgChange(id);
    props.handleContextChange();
  };

  const claimOrg = async (active) => {
    props.handleContextChange();
    props.handleInputChange('');
    await props.handleOrgConfirm(active, props.relay.environment);
    props.handleOrgChange(null);
  };

  const cancelClaim = () => {
    props.handleOrgChange(null);
    props.handleContextChange();
    props.handleInputChange('');
  };

  const handleAddOrganisation = async (variables) => {
    const {environment} = props.relay;
    const response = await createOrganisationMutation(environment, variables);
    console.log(response);
    const {operatorName, rowId} = response.createOrganisation.organisation;
    selectOrg(operatorName, rowId);
  };

  const {edges} = session.ciipUserBySub.ciipUserOrganisationsByUserId;
  return (
    <>
      {props.flagCertRequests && (
        <Alert variant="info">
          One or more reporting operations has requested that you certify an
          application.{' '}
          <Alert.Link href="/certifier/requests">
            View all certification requests.
          </Alert.Link>
        </Alert>
      )}
      <Card bg="light">
        <Card.Body>
          Operator, Operator Representative, and Reporting Operation are defined
          in the{' '}
          <a href="http://www.bclaws.ca/civix/document/id/complete/statreg/14029_01">
            Greenhouse Gas Industrial Reporting and Control Act
          </a>{' '}
          and{' '}
          <a href="http://www.bclaws.ca/civix/document/id/lc/statreg/249_2015">
            Greenhouse Gas Emission Reporting Regulation
          </a>
          .
        </Card.Body>
      </Card>
      <br />
      {edges.length > 0 && (
        <Table
          striped
          bordered
          hover
          style={{textAlign: 'left', marginBottom: '40px'}}
        >
          <thead>
            <tr>
              <th>Operator</th>
              <th>Access Status</th>
              <th>Operations/Facilities</th>
            </tr>
          </thead>
          <tbody>
            {edges.map(({node}) => {
              return <UserOrganisation key={node.id} userOrganisation={node} />;
            })}
          </tbody>
          <style jsx>
            {`
              th {
                padding: 20px;
                font-weight: 500;
              }
            `}
          </style>
        </Table>
      )}

      <div>
        <Card style={{marginTop: '50px'}}>
          <Card.Body>
            {props.confirmOrg ? (
              <>
                <Card.Title>Requesting access to: </Card.Title>
                <h4 style={{fontWeight: 300, margin: '15px 0'}}>
                  {props.orgInput}{' '}
                </h4>
                {/* Dev-only button to automatically create approved user-organisation requests */}
                {process.env.NODE_ENV === 'production' ? null : (
                  <Button
                    style={{marginRight: '15px'}}
                    variant="success"
                    onClick={async () => claimOrg(true)}
                  >
                    Activate Access
                  </Button>
                )}
                <Button
                  style={{marginRight: '15px'}}
                  variant="primary"
                  onClick={async () => claimOrg(false)}
                >
                  Request Access
                </Button>
                <Button variant="danger" onClick={cancelClaim}>
                  Cancel
                </Button>
              </>
            ) : (
              <Dropdown className="search-dropdown">
                <Card.Title>
                  Request access to apply for the CIIP on behalf of a Reporting
                  Operation:
                </Card.Title>
                <Alert variant="warning">
                  <p>
                    The CleanBC Industrial Incentive Program application must be
                    submitted by the Operator of the Reporting Operation or, if
                    there is more than one Operator, the Designated Operator as
                    defined by the{' '}
                    <Alert.Link href="http://www.bclaws.ca/civix/document/id/complete/statreg/14029_01">
                      Greenhouse Gas Industrial Reporting and Control Act
                    </Alert.Link>{' '}
                    and{' '}
                    <Alert.Link href="http://www.bclaws.ca/civix/document/id/lc/statreg/249_2015">
                      Greenhouse Gas Emission Reporting Regulation
                    </Alert.Link>
                    . Other representatives of the Reporting Operation may
                    complete the application if authorization from the Operator
                    is provided.
                  </p>
                  <p>
                    Further information on the role of the Operator and
                    Certifying Official{' '}
                    <Alert.Link href="https://www2.gov.bc.ca/gov/content/environment/climate-change/industry/cleanbc-program-for-industry/cleanbc-industrial-incentive-program">
                      can be found here
                    </Alert.Link>
                    .
                  </p>
                  <p>
                    Please email{' '}
                    <Alert.Link href="mailto:GHGRegulator@gov.bc.ca?subject=CIIP Portal Inquiry">
                      GHGRegulator@gov.bc.ca
                    </Alert.Link>{' '}
                    if you have any questions.
                  </p>
                </Alert>
                <small style={{display: 'block', marginBottom: '20px'}}>
                  (You can search to narrow the results in the dropdown)
                </small>
                <Dropdown.Toggle id="org-dropdown" className="search-toggle">
                  Select Operator
                </Dropdown.Toggle>
                <Dropdown.Menu>
                  <FormControl
                    placeholder="Search..."
                    className="mx-3 my-2 w-auto"
                    onChange={changeInput}
                  />
                  <div className="org-scroll">
                    {allOrganisations.edges.map(({node}) => {
                      return (
                        <Organisation
                          key={node.id}
                          select
                          organisation={node}
                          orgInput={props.orgInput}
                          selectOrg={selectOrg}
                        />
                      );
                    })}
                  </div>
                </Dropdown.Menu>
              </Dropdown>
            )}
            <hr />
            <AddOrganisationFacility
              onAddOrganisation={handleAddOrganisation}
            />

            <style jsx>
              {`
                .org-scroll {
                  max-height: 250px;
                  overflow: hidden;
                  overflow-y: scroll;
                }
              `}
            </style>
          </Card.Body>
        </Card>
      </div>
    </>
  );
};

export default createFragmentContainer(OrganisationsComponent, {
  query: graphql`
    fragment Organisations_query on Query {
      session {
        ciipUserBySub {
          id
          ciipUserOrganisationsByUserId(first: 2147483647)
            @connection(key: "Organisations_ciipUserOrganisationsByUserId") {
            edges {
              node {
                id
                ...UserOrganisation_userOrganisation
              }
            }
          }
        }
      }

      allOrganisations {
        edges {
          node {
            id
            ...Organisation_organisation
          }
        }
      }
    }
  `
});
