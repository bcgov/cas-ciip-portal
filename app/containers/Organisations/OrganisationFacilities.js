import React from 'react';
import {graphql, createFragmentContainer} from 'react-relay';
import {CardDeck} from 'react-bootstrap';
import LoadingSpinner from '../../components/LoadingSpinner';
import Facility from './Facility';

export const OrganisationFacilitiesComponent = props => {
  const {organisation} = props.query;
  if (!organisation) return <LoadingSpinner />;

  return (
    <>
      <CardDeck>
        {organisation.facilitiesByOrganisationId.edges.map(({node}) => {
          return <Facility key={node.id} facility={node} />;
        })}
      </CardDeck>
    </>
  );
};

export default createFragmentContainer(OrganisationFacilitiesComponent, {
  query: graphql`
    fragment OrganisationFacilities_query on Query
      @argumentDefinitions(id: {type: "ID!"}) {
      organisation(id: $organisationId) {
        id
        facilitiesByOrganisationId(first: 2147483647)
          @connection(key: "organisation_facilitiesByOrganisationId") {
          edges {
            node {
              id
              ...Facility_facility
            }
          }
        }
      }
    }
  `
});
