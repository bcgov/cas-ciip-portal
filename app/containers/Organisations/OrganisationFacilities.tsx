import React, {useEffect} from 'react';
import {graphql, createRefetchContainer} from 'react-relay';
import {Table, Container, Row, Col} from 'react-bootstrap';
import SortableTableHeader from 'components/SortableTableHeader';
import SearchBox from 'components/SearchBox';
import LoadingSpinner from '../../components/LoadingSpinner';
import Facility from './Facility';

export const OrganisationFacilitiesComponent = props => {
  const {
    organisationRowId,
    searchDisplay,
    direction,
    orderByField,
    searchField,
    searchValue,
    handleEvent
  } = props;
  const {edges} = props.query.searchOrganisationFacilities;
  console.log(organisationRowId);

  useEffect(() => {
    const refetchVariables = {
      organisationRowId,
      searchField,
      searchValue,
      orderByField,
      direction
    };
    props.relay.refetch(refetchVariables);
  });

  const tableHeaders = [
    {columnName: 'facility_name', displayName: 'Facility Name'},
    {columnName: 'address', displayName: 'Address'},
    {columnName: 'application_status', displayName: 'Status'}
  ];

  const dropdownSortItems = ['Facility Name', 'Address', 'Status'];

  const displayNameToColumnNameMap = {
    'Facility Name': 'facility_name',
    Address: 'address',
    Status: 'application_status'
  };

  const facilities = props.query.searchOrganisationFacilities;
  if (!facilities) return <LoadingSpinner />;

  return (
    <>
      {/* <CardDeck>
        {organisation.facilitiesByOrganisationId.edges.map(({node}) => {
          return <Facility key={node.id} facility={node} />;
        })}
      </CardDeck> */}
      <Container>
        <Row>
          <Col md={{span: 12, offset: 6}}>
            <SearchBox
              dropdownSortItems={dropdownSortItems}
              handleEvent={handleEvent}
              displayNameToColumnNameMap={displayNameToColumnNameMap}
              searchDisplay={searchDisplay}
            />
          </Col>
        </Row>
      </Container>

      <Table striped bordered hover style={{textAlign: 'center'}}>
        <thead>
          <tr>
            {tableHeaders.map(header => (
              <SortableTableHeader
                key={header.columnName}
                sort={handleEvent}
                headerVariables={header}
              />
            ))}
            <th />
          </tr>
        </thead>
        <tbody>
          {edges.map(edge => (
            <Facility key={edge.node.id} facility={edge.node} />
          ))}
        </tbody>
      </Table>
    </>
  );
};

export default createRefetchContainer(
  OrganisationFacilitiesComponent,
  {
    query: graphql`
      fragment OrganisationFacilities_query on Query
        @argumentDefinitions(
          organisationRowId: {type: "String!"}
          searchField: {type: "String"}
          searchValue: {type: "String"}
          orderByField: {type: "String"}
          direction: {type: "String"}
        ) {
        searchOrganisationFacilities(
          organisationRowId: $organisationRowId
          searchField: $searchField
          searchValue: $searchValue
          orderByField: $orderByField
          direction: $direction
        ) {
          edges {
            node {
              id
              ...Facility_facility
            }
          }
        }
      }
    `
  },
  graphql`
    query OrganisationFacilitiesRefetchQuery(
      $organisationRowId: String!
      $searchField: String
      $searchValue: String
      $orderByField: String
      $direction: String
    ) {
      query {
        ...OrganisationFacilities_query
          @arguments(
            organisationRowId: $organisationRowId
            orderByField: $orderByField
            direction: $direction
            searchField: $searchField
            searchValue: $searchValue
          )
      }
    }
  `
);
