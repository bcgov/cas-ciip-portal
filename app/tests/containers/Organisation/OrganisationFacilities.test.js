import React from 'react';
import {shallow} from 'enzyme';
import {OrganisationFacilitiesComponent} from '../../../containers/Organisations/OrganisationFacilities';

describe('OrganisationFacilities', () => {
  it('should render no OrganisationFacilities if the organisation has no facilities', async () => {
    const query = {
      organisation: {
        id: 'id',
        facilitiesByOrganisationId: {edges: []}
      }
    };
    const r = shallow(<OrganisationFacilitiesComponent query={query} />);
    expect(r).toMatchSnapshot();
  });
  it("should render the organisation's facilities", async () => {
    const query = {
      organisation: {
        id: 'id',
        facilitiesByOrganisationId: {
          edges: [
            {
              node: {
                id: 'FacId'
              }
            }
          ]
        }
      }
    };
    const r = shallow(<OrganisationFacilitiesComponent query={query} />);
    expect(r).toMatchSnapshot();
    expect(r.find('Relay(FacilityComponent)').prop('facility')).toBe(
      query.organisation.facilitiesByOrganisationId.edges[0].node
    );
  });
});
