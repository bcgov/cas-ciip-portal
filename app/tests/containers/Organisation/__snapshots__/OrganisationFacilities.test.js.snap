// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationFacilities should render no OrganisationFacilities if the organisation has no facilities 1`] = `
<Fragment>
  <CardDeck />
</Fragment>
`;

exports[`OrganisationFacilities should render the organisation's facilities 1`] = `
<Fragment>
  <CardDeck>
    <Relay(FacilityComponent)
      facility={
        Object {
          "id": "FacId",
        }
      }
      key="FacId"
    />
  </CardDeck>
</Fragment>
`;
